swagger_server/itm/itm_scene.py

```diff
@@ -167,6 +167,9 @@ def action_taken(self, action :Action, session_state :State):
                 # Respond to probes if conditions are met.
                 if self.conditions_met(mapping.conditions, session_state, mapping.condition_semantics):
                     self.parent_scenario.respond_to_probe(mapping.probe_id, mapping.choice, action.justification)
+                # If action doesn't respond to probe, set probe response in meta info to None
+                if self.parent_scenario.session.kdma_training:
+                    self.parent_scenario.session.state.meta_info.probe_response = None
```

Comment:
You are 100% right...

Reply:
I didn't actually *test* my `elif` suggestion...it turns out it's wrong too.   :)   That code only gets executed if we found an action mapping-- not if they select one of the unrestricted actions that isn't an action mapping.  So it was keeping the old probe responses until the ADM selected an action from the configured action mappings.

I think it needs to set it to None before the `for` loop.  Then if it ends up firing off a probe response, it will get added to the meta info.